### PR TITLE
Fix LoadBalancer externalTrafficPolicy:Local issue on upgrade to v0.11

### PR DESCRIPTION
**Pull request by BSWANG**
## Description

Remove iptables create by previous version which without `--ramdom-fully`
If not remove the iptables rules. The `externalTrafficPolicy:Local` will can not reserve the client IP.

```
Chain POSTROUTING (policy ACCEPT 1 packets, 60 bytes)
 pkts bytes target     prot opt in     out     source               destination
  283 20269 KUBE-POSTROUTING  all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* kubernetes postrouting rules */
    0     0 MASQUERADE  all  --  *      !docker0  169.254.123.0/24     0.0.0.0/0
   12   771 MASQUERADE  all  --  *      *       172.30.0.0/16       !224.0.0.0/4
    0     0 MASQUERADE  all  --  *      *      !172.30.0.0/16        172.30.0.0/16
    0     0 RETURN     all  --  *      *       172.30.0.0/16        172.30.0.0/16
    0     0 MASQUERADE  all  --  *      *       172.30.0.0/16       !224.0.0.0/4          random-fully
    0     0 RETURN     all  --  *      *      !172.30.0.0/16        172.30.0.128/25
    0     0 MASQUERADE  all  --  *      *      !172.30.0.0/16        172.30.0.0/16        random-fully
```

## Todos
- [x] Tests
- [x] Documentation
- [x] Release note (need review)

## Release Note

```release-note
Maybe: Fix `externalTrafficPolicy: Local` service can not reserve client ip after upgrade flannel.
```

### CODE REVIEW

**Comment by anton-johansson** [comment]
Any news on this? We're running `flannel` version `0.11.0` and I'm trying to introduce MetalLB in front of our NGINX pods to make sure we can see the real IP address, but it doesn't seem to work. I stumbled over this issue which seems to be related.

**Comment by BSWANG** [comment]
@rakelkar @SleepyBrett @tomdee

**Comment by rajatchopra** [comment]
@BSWANG Looks useful. Please rebase.

**Comment by BSWANG** [comment]
@rajatchopra 
Rebased.

**Comment by manuelbuil** [comment]
Hey, sorry for not paying attention to this PR. Could you rebase again? After that, I'll review and hopefully merge it!

**Comment by BSWANG** [comment]
@manuelbuil Done!

File: network/iptables.go [hunk 1]
```diff
@@ -11,6 +11,7 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+//go:build !windows
```
Comment (manuelbuil) [inline network/iptables.go:14]: As we already have line 15, does this line bring any value?

File: network/iptables_test.go [hunk 2]
```diff
@@ -11,6 +11,7 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+//go:build !windows
```
Comment (manuelbuil) [inline network/iptables_test.go:14]: same comment

File: network/iptables.go [hunk 3]
```diff
@@ -202,19 +254,40 @@ func DeleteIP6Tables(rules []IPTablesRule) error {
...
-	if exists {
+	ensureDelete := make([]IPTablesRule, 0)
+	for _, ruleToDelete := range rulesToDelete {
+		ruleExist := false
+		for _, rule := range rules {
+			if rule.Equal(&ruleToDelete) {
+				ruleExist = true
+				break
+			}
+		}
```
Comment (manuelbuil) [inline network/iptables.go:270]: What happens if we try to delete them and they don't exist? I wonder if that might be better than checking if they exist or not

Comment (BSWANG) [reply network/iptables.go:270]: OK, Added rule check before add to rulelist.

File: network/iptables.go (hunk 1 above)
Comment (BSWANG) [reply network/iptables.go]: Remove the useless `go:build` line which added by goland automatic.

**Comment by manuelbuil** [comment]
I was trying to reproduce the issue but I can't see those iptable rules. Are they only being created before v0.11?

**Comment by BSWANG** [comment]
@manuelbuil Yes, those iptables rules created without 'random-fully' before v0.11. And not got cleanup when upgrade to v0.11+.

**Comment by manuelbuil** [comment]
> @manuelbuil Yes, those iptables rules created without 'random-fully' before v0.11. And not got cleanup when upgrade to v0.11+.

In that case, I wonder if we should merge this. This only affects very very old versions and we can't backport. Therefore, it will onlyaffect users moving from v0.10 to v0.15, which is really strange, right?

**Comment by luthermonson** [comment]
this is for such an older release... i hate to say this but maybe this is stale now and we should just close it. unfortunately maybe just a clean install is required now and we'll have to live with that. reopen if you have a problem with that @BSWANG

**Pull request closed without merging**